### PR TITLE
SAMZA-1196: Fix TestJmxReporter

### PR DESCRIPTION
**Pull request by shanthoosh**

### CODE REVIEW

**Comment by shanthoosh** [comment]
@prateekm @navina 

Please take a look when you get chance.

**Comment by shanthoosh** [comment]
@prateekm 

Added additional asserts to test metrics reporting through listener. 

Thanks.

File: samza-core/src/test/scala/org/apache/samza/metrics/reporter/TestJmxReporter.scala [hunk 1]
```diff
@@ -19,65 +19,49 @@
...
-    val mbserver = JMXConnectorFactory.connect(url).getMBeanServerConnection
-    val stateViaJMX = mbserver.getAttribute(new ObjectName("org.apache.samza.metrics.JvmMetrics:type=test,name=mem-non-heap-used-mb"), "Value").asInstanceOf[Float]
+    verify(mBeanServerMock, times(1)).registerMBean(Matchers.anyObject(), Matchers.eq(objectName))
+
+    reset(mBeanServerMock)
+    // Create dummy counter to test metrics reporting through listener.
+    registry.newCounter(metricGroup, metricName)
 
-    assertTrue(stateViaJMX > 0)
+    assertEquals(1, registry.listeners.size)
```
Comment (prateekm) [inline samza-core/src/test/scala/org/apache/samza/metrics/reporter/TestJmxReporter.scala]: Not necessary, this is registry internal state.

**Comment by navina** [comment]
lgtm! Thanks for fixing this 👍

**Pull request closed without merging**